keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -514,310 +481,126 @@ spec:
-one addition being that each named resource can have a set of arbitrary
-attributes attached to it.
-
-If a driver wanted to use a different structured model to represent its resources,
-a new structured model would need to be defined inside Kuberenetes, and a field
-would need to be added to this struct at the same level as
-`namedResources`. Driver implementors would then have the option
-to set this new field instead.
-
-**Note:** If a new model is added to the schema but clients are not updated,
-they'll encounter an object with no information from any known structured model
-when they serialize into their known version of a `ResourceSlice`. This
-tells them that they cannot handle the object because the API has been extended.
-
-Drivers can use different structured models by publishing multiple
-`ResourceSlice` objects, as long as each model represents a distinct set of
-resources. Whether the information about resources of one particular structured
-model must fit into one ResourceSlice object (or be distributed across
-many) depends on how that particular structured model describes its resources. In
-all cases, the size of each object is a hard limit and one must take this into
-account when designing a structured model and preparing ResourceSlice objects
-for it.
-
-Below is an example of a driver that provides two discrete GPU cards using the
-`namedResources` model described above:
+In the kubelet case, each driver running on a node publishes a set of
+`ResourceSlice` objects to the API server for its own resources, using its
+connection to the apiserver. The collection of these objects form a pool from
+which resources can be allocated. Some additional fields (defined in the API
+section) enable a consumer to determine whether it has a complete and
+consistent view of that pool.
+
+Access control through a validating admission
+policy can ensure that the drivers running on one node are not allowed to
+create or modify `ResourceSlices` belonging to another node. The `nodeName`
+and `driverName` fields in each `ResourceSlice` object are used to determine which objects are
+managed by which driver instance. The owner reference ensures that objects
+beloging to a node get cleaned up when the node gets removed.
+
+In addition, whenever kubelet starts, it first deletes all `ResourceSlices`
+belonging to the node with a `DeleteCollection` call that uses the node name in
+a field filter. This ensures that no pods depending in DRA get scheduled to the
+node until the required DRA drivers have started up again (node reboot) and
+reconnected to kubelet (kubelet restart). It also ensures that drivers which
+don't get started up again at all don't leave stale `ResourceSlices`
+behind. Garbage collection does not help in this case because the node object
+still exists. For the same reasons, the ResourceSlices belonging to a driver
+get removed when the driver unregisters, this time with a field filter for node
+name and driver name.
```

Comment:
As mentioned elsewhere, does this then imply that we will never be able to store a `status` on these objects?

Reply:
We can add a status. Once we do, we need to become more careful with deletion and add more rules around it.

Reply:
Right, it's on the cusp. Someone still needs to define what that status looks like and who populates it (https://github.com/kubernetes-sigs/wg-device-management/issues/19).

I prefer to merge this without a status, just to be sure that we have something that is agreed upon for 1.31. Perhaps we can get an extension to add the status.

Reply:
I looked into this today (see issue comments), but it's not really obvious what the exact use cases are and how to model them in the API. Let's not try for 1.31.